admin/src/components/dashboard.js

```diff
@@ -0,0 +1,220 @@
+import React, { Component } from 'react';
+
+import BarList from './barList';
+import BarDetails from './barDetails';
+
+class Dashboard extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      ownerData: {},
+      activeBar: null,
+      staffCode: null,
+    };
+  }
+
+  getOwnerData = () => {
+    const { token } = this.props;
+    return fetch('/owner/me',
+      {
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      })
+      .then(res => res.json())
+      .then(res => this.setState({ ownerData: res.user }));
+  }
+
+  addBar = (event, barObj) => {
+    event.preventDefault();
+    const { token } = this.props;
+    fetch('/owner/bars',
```

Comment:
I see some code repetition regarding http requests (both with fetch an axios). Try to centralize this logic in some place, if you don't want a redux middleware you can create a JS module that exports an instance of a class that will be shared.

Also, take advantage of that shared instance for sharing some data (token) that will be appended automatically into the headers.

Reply:
Imagine this code as:

```js
addBar = (event, barObj) => {
    event.preventDefault();
    apiClient.addBar(barObj)
      .then(res => this.setState({ ownerData: res }));
};
```

Reply:
![](https://www.mememaker.net/api/bucket?path=static/img/memes/full/2014/Oct/21/5/it-feels-so-good3.jpg)